public/js/reducers/sources.js

```diff
@@ -224,15 +215,15 @@ function getSourceText(state: OuterState, id: string) {
 }
 
 function getSourceTabs(state: OuterState) {
-  return state.sources.tabs;
+  const tabs = state.sources.tabs;
+  return tabs
+    .filter(url => getSourceByURL(state, url))
+    .map(url => getSourceByURL(state, url));
```

Comment:
I took "url" from the navigate/will-navigate event payload from the tabTarget. But more importantly, this would only work for Firefox. I'll look to see if I can do something for chrome.